src/rpcblockchain.cpp

```diff
@@ -665,6 +666,14 @@ UniValue getblockchaininfo(const UniValue& params, bool fHelp)
             "        },\n"
             "        \"reject\": { ... }      (object) progress toward rejecting pre-softfork blocks (same fields as \"enforce\")\n"
             "     }, ...\n"
+            "  ],\n"
+            "  \"upgrades\": [            (array) status of non-forward compatible upgrades\n"
+            "     {\n"
+            "        \"upgradeid\": \"xxxx\",        (string) name of upgrade\n"
+            "        \"status\": xxxx,         (string) status of upgrade\n"
+            "        \"height\": xxxxxx,         (numeric) block height of activation\n"
+            "        \"info\": xxxx,         (string) additional information about upgrade\n"
```

Comment:
What would this additional information about the upgrade be used for? The conversation above suggests putting the name of the upgrade (i.e. "Overwinter"). While I think we should put a colloquial name for it somewhere, I feel like we need more fields to include some of the additional information we wanted to communicate, recall https://github.com/zcash/zcash/issues/2454#issuecomment-348248624, we wanted to communicate: 

1) meta-info about the upgrade, "an upgrade called x is happening at y"
2) the specifics of the upgrade, "this upgrade does this and that and that"
3) the message that wallets/other GUIs should present it to the users so it can be consistent 

I think we have done a good job with 1) so far, but not 2) and 3). To me, the info field makes sense for 2), and we should create a field to contain 3).

Reply:
Info field was intended for 2, how about something like "NU0 - this upgrade performs a non-backwards compatible network upgrade to add features in preparation for Sapling"

Reply:
Can we just suggest that wallets and exchanges display info message (above) to the users?